src/token_request_handler.ts

```diff
@@ -46,6 +51,26 @@ export class BaseTokenRequestHandler implements TokenRequestHandler {
     return (response as TokenErrorJson).error === undefined;
   }
 
+  performRevokeTokenRequest(
+      configuration: AuthorizationServiceConfiguration,
+      request: RevokeTokenRequest): Promise<boolean> {
+    let revokeTokenResponse = this.requestor.xhr<boolean>({
+      url: configuration.revocationEndpoint,
+      method: 'POST',
+      dataType: 'json',  // adding implicit dataType
+      headers: {'Content-Type': 'application/x-www-form-urlencoded'},
+      data: this.utils.stringify(request.toStringMap())
+    });
+
+    return revokeTokenResponse
+        .then(response => {
+          return true;
+        })
+        .catch(error => {
```

Comment:
You don't need to catch and rethrow. Just remove the catch block. The promise chain will get rejected anyway.